internal/environments/environments.go

```diff
@@ -0,0 +1,60 @@
+// EnvironmentController, public interface for Environment objects
+
+// Package environments defines data structure representing
+// a environment instance and methods for interacting with them
+// it is left to concrete implementations in package db or others to
+// implement these interfaces
+package environments
+
+import (
+	"errors"
+
+	"gorm.io/gorm"
+)
+
+// EnvironmentController is the management layer for environments
+type EnvironmentController struct {
+	store environmentStore
+	// store jobRequestStore // TODO: placeholder for other store environment controller can touch
```

Comment:
Small note if we ever need to give the environements controller access to other functionality, ie services. It might be better to embed the `ServiceController` rather than the `serviceStore`. That means we can only uses Services public API from other packages. This forces all the underlying logic for doing things related to services to stay in package services.

Reply:
^ Is what I did for builds to give it the ability to create a new service if it recieves a build request for a service that doesn't exist already.